sonar-scanner-engine/src/main/java/org/sonar/scanner/genericcoverage/GenericCoverageReportParser.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.scanner.genericcoverage;
+
+import com.google.common.base.Preconditions;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.xml.stream.XMLStreamException;
+import org.codehaus.staxmate.in.SMHierarchicCursor;
+import org.codehaus.staxmate.in.SMInputCursor;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.coverage.NewCoverage;
+import org.sonar.api.utils.StaxParser;
+
+public class GenericCoverageReportParser {
+
+  private static final String LINE_NUMBER_ATTR = "lineNumber";
+  private static final String COVERED_ATTR = "covered";
+  private static final String BRANCHES_TO_COVER_ATTR = "branchesToCover";
+  private static final String COVERED_BRANCHES_ATTR = "coveredBranches";
+
+  private static final int MAX_STORED_UNKNOWN_FILE_PATHS = 5;
+
+  private int numberOfUnknownFiles;
+  private final List<String> firstUnknownFiles = new ArrayList<>();
+  private final Set<String> matchedFileKeys = new HashSet<>();
+
+  public void parse(java.io.File reportFile, SensorContext context) {
+    try (InputStream inputStream = new FileInputStream(reportFile)) {
+      parse(inputStream, context);
+    } catch (Exception e) {
+      throw new IllegalStateException("Error during parsing of coverage report " + reportFile, e);
+    }
+  }
+
+  void parse(InputStream inputStream, SensorContext context) throws XMLStreamException {
+    new StaxParser(rootCursor -> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "StaxParser"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)